src/OptionsResolver.php

```diff
@@ -0,0 +1,401 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver as BaseOptionsResolver;
+
+/**
+ * @internal
+ */
+final class OptionsResolver extends BaseOptionsResolver
```

Comment:
Another option would be not to inherit, but to compose.

Quick draft:

```php
final class FixerConfigurationResolver
{
    private $options; // FixerOptions[]
    private $optionsResolver; // Symfony class instance we use in following methods

    public function setRoot(string $name) { ... } // to be removed on 3.x line
    public function setOption(FixerOption $option) { ... }
    public function getOptions() :FixerOption[] { ... }
    public function resolve(array $opts) :array {...}
}

final class FixerOption // just a DTO
{
// required
    private $name;
    private $description;

// optional
    private $default;
    private $normalizer;
    private $validator;

    // getters, setters
}
```

that way we do not modify or rely on internal Symfony code.